### PR TITLE
Cleanup of dead code

### PR DESCRIPTION
**Pull request by dbussink**
Go recently announced https://go.dev/blog/deadcode and this does a first pass and cleans up a bunch of things.

Not everything it identifies is really dead code for us. We have a bunch of test implementations of interfaces for example, which is seen as dead code when running without the `-test` flag.

But with the `-test` flag, it doesn't find code that only is used from tests.

What it points at is a good starting point though for what to then manually verify and clean up. That's what I did here.

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported to release branches
-   [x] If this change is to be back-ported to previous releases, a justification is included in the PR description
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on CI?
-   [x] Documentation was added or is not required

### CODE REVIEW

**Review by ajm188 — APPROVED** [review]

File: go/vt/schemadiff/schema_test.go [hunk 1]
```diff
@@ -162,7 +162,7 @@ func TestNewSchemaFromQueriesLoop(t *testing.T) {
 	)
 	_, err := NewSchemaFromQueries(queries)
 	require.Error(t, err)
-	err = vterrors.UnwrapFirst(err)
+	err = errors.Unwrap(err)
```
Comment (frouioui) [inline go/vt/schemadiff/schema_test.go:165]: I think this change might be causing the issue in the unit test CI:
```
=== RUN   TestNewSchemaFromQueriesLoop
    schema_test.go:166: 
        	Error Trace:	/home/runner/work/vitess/vitess/go/vt/schemadiff/schema_test.go:166
        	Error:      	An error is expected but got nil.
        	Test:       	TestNewSchemaFromQueriesLoop
--- FAIL: TestNewSchemaFromQueriesLoop (0.00s)
```

**Review by frouioui — APPROVED** [review]

File: go/vt/schemadiff/schema_test.go (hunk 1 above)
Comment (dbussink) [reply go/vt/schemadiff/schema_test.go:165]: Whoops, didn't intend to commit this change from some testing I did.

**Review by harshit-gangal — APPROVED** [review]
nice clean up 💯

File: go/vt/schemadiff/schema_test.go (hunk 1 above)
Comment (frouioui) [reply go/vt/schemadiff/schema_test.go]: Got it!

**Pull request merged**